lib/miq_expression.rb

```diff
@@ -600,12 +600,20 @@ def self.value2human(val, options = {})
 
   UNQUOTABLE_OPERATORS = %w[= IS\ NULL IS\ NOT\ NULL IS\ EMPTY IS\ NOT\ EMPTY].freeze
 
+  def self.quote_by(operator, value, column_type = nil)
+    if (STRING_OPERATORS + DEPRECATED_OPERATORS - UNQUOTABLE_OPERATORS).map(&:downcase).include?(operator)
+      value
+    else
+      quote(value, column_type.to_s)
+    end
+  end
+
   def self.operands2rubyvalue(operator, ops, context_type)
     # puts "Enter: operands2rubyvalue: operator: #{operator}, ops: #{ops.inspect}"
 
     if ops["field"]
       if ops["field"] == "<count>"
-        ["<count>", quote(ops["value"], "integer")]
+        ["<count>", quote_by(nil, ops["value"], :integer)]
```

Comment:
one of these is `"string"` and the other is `:integer` - is this intentional?
Since you're doing a `to_s`, could you pass in `"integer"` here instead?